boards/arm/mps2/CMakeLists.txt

```diff
@@ -1,18 +1,13 @@
-#
-# Copyright (c) 2019,2020 Linaro Limited
-#
 # SPDX-License-Identifier: Apache-2.0
-#
 
 zephyr_library()
 zephyr_library_sources(pinmux.c)
 
-if(CONFIG_SOC_MPS2_AN521_CPU1 AND NOT CONFIG_OPENAMP)
-# Building a firmware image for CPU1: this requires a binary
-# for CPU0, which will boot the device and wake up CPU1.
-# However, if building with OPENAMP, there is no need to build
-# any binary for CPU0, as this is built by the dual core sample.
-
+if(CONFIG_BOARD_MPS2_AN521_CPU1 AND NOT CONFIG_OPENAMP)
```

Comment:
Not wrong, but wouldn't it be better to check for the SoC itself ?

Often users copy board code, and in case of testing for a SoC the copied code would still work if SoC is identical whereas a board could end-up not working.

Reply:
No, this code specifically checks for the second CPU and adds a dummy second image for the first CPU so that the second CPU can be booted, it would make no sense to check for the soc here